File: src/main/java/com/kurtraschke/gtfsrtdump/output/JsonOutput.java
package com.kurtraschke.gtfsrtdump.output;

import com.fasterxml.jackson.databind.ObjectMapper;
import com.google.transit.realtime.GtfsRealtime;
import com.hubspot.jackson.datatype.protobuf.ProtobufModule;
import picocli.CommandLine.Command;

import java.io.IOException;

@Command(name = "json", description = "JSON output")
public class JsonOutput extends OutputMethod {

    @Override
    public Integer format(GtfsRealtime.FeedMessage fm) throws IOException {
        final ObjectMapper mapper = new ObjectMapper();
        mapper.registerModule(new ProtobufModule());

        mapper.writerWithDefaultPrettyPrinter().writeValue(System.out, fm);

        return 0;
    }
}

File: src/main/kotlin/com/kurtraschke/gtfsrtdump/output/JsonOutput.kt
package com.kurtraschke.gtfsrtdump.output

import com.fasterxml.jackson.databind.ObjectMapper
import com.google.transit.realtime.GtfsRealtime.FeedMessage
import com.hubspot.jackson.datatype.protobuf.ProtobufJacksonConfig
import com.hubspot.jackson.datatype.protobuf.ProtobufModule
import picocli.CommandLine.Command
import java.io.PrintWriter

@Command(name = "json", description = ["JSON output"])
class JsonOutput : OutputMethod() {

    override fun format(fm: FeedMessage, w: PrintWriter): Int {
        val config = ProtobufJacksonConfig.builder().extensionRegistry(main.registry).build()

        val mapper = ObjectMapper()
        mapper.registerModule(ProtobufModule(config))
        mapper.writerWithDefaultPrettyPrinter().writeValue(w, fm)
        return 0
    }

}
